### ISSUE TITLE
bitcoind: "FileCommit: fcntl F_FULLFSYNC failed" on MacOS Ventura

### ISSUE THREAD

**Issue by asylejmani** [bitcoin/bitcoin#26650]
Running bitocind in newest MacOS Ventura crashes randomly, but not on previous MacOS versions. I can confirm it works on my other MBP (Monterey) as I managed to download the whole chain without any issues.

**Expected behavior**
To keep syncing blocks

**Actual behavior**
Fails at random blocks for Ventura.

**To reproduce**
Try syncing the chain either via bitcoin gui or bitcoin cli on MacOS Ventura!

**System information**
Machine: Apple MBP M1 16GB MacOS Ventura 13.0.1
Version: 22, 23 and 24 (downloaded from the website)
Samsung External T7 SSD (I can confirm the device works fine as I downloaded the whole chain in it with the other MBP)

```2022-12-06T21:04:05Z UpdateTip: new best=00000000000000000003c366fb85d272bf612692e1798f5473550f08dd166480 height=765910 version=0x31c5c000 log2_work=93.878232 tx=785535083 date='2022-12-04T22:21:16Z' progress=0.999374 cache=84.2MiB(631034txo)
2022-12-06T21:04:05Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:05Z ERROR: Flush: failed to commit file 3302
2022-12-06T21:04:05Z *** Flushing block file to disk failed. This is likely the result of an I/O error.
2022-12-06T21:04:05Z Error: A fatal internal error occurred, see debug.log for details
2022-12-06T21:04:05Z tor: Thread interrupt
2022-12-06T21:04:05Z Shutdown: In progress...
2022-12-06T21:04:05Z torcontrol thread exit
2022-12-06T21:04:05Z opencon thread exit
2022-12-06T21:04:05Z addcon thread exit
2022-12-06T21:04:05Z net thread exit
2022-12-06T21:04:06Z UpdateTip: new best=00000000000000000003029c5ff69ac0441d3ca181626f9b895493af4a13bc40 height=765911 version=0x236ca000 log2_work=93.878244 tx=785535847 date='2022-12-04T22:24:48Z' progress=0.999375 cache=84.3MiB(631479txo)
2022-12-06T21:04:06Z msghand thread exit
2022-12-06T21:04:06Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:06Z ERROR: SerializeFileDB: Failed to flush file /Volumes/Bitcoin/peers.adeb
2022-12-06T21:04:06Z DumpAnchors: Flush 2 outbound block-relay-only peer addresses to anchors.dat started
2022-12-06T21:04:06Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:06Z ERROR: SerializeFileDB: Failed to flush file /Volumes/Bitcoin/anchors.256c
2022-12-06T21:04:06Z DumpAnchors: Flush 2 outbound block-relay-only peer addresses to anchors.dat completed (0.00s)
2022-12-06T21:04:06Z scheduler thread exit
2022-12-06T21:04:06Z Writing 0 unbroadcast transactions to disk.
2022-12-06T21:04:06Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:06Z Failed to dump mempool: FileCommit failed. Continuing anyway.
2022-12-06T21:04:06Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:06Z ERROR: Flush: failed to commit file 3303
2022-12-06T21:04:06Z *** Flushing block file to disk failed. This is likely the result of an I/O error.
2022-12-06T21:04:06Z Error: A fatal internal error occurred, see debug.log for details
2022-12-06T21:04:06Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:06Z ERROR: Flush: failed to commit file 3303
2022-12-06T21:04:06Z *** Flushing undo file to disk failed. This is likely the result of an I/O error.
2022-12-06T21:04:06Z Error: A fatal internal error occurred, see debug.log for details
2022-12-06T21:04:07Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:07Z ERROR: Flush: failed to commit file 3303
2022-12-06T21:04:07Z *** Flushing block file to disk failed. This is likely the result of an I/O error.
2022-12-06T21:04:07Z Error: A fatal internal error occurred, see debug.log for details
2022-12-06T21:04:07Z FileCommit: fcntl F_FULLFSYNC failed: 25
2022-12-06T21:04:07Z ERROR: Flush: failed to commit file 3303
2022-12-06T21:04:07Z *** Flushing undo file to disk failed. This is likely the result of an I/O error.
2022-12-06T21:04:07Z Error: A fatal internal error occurred, see debug.log for details
2022-12-06T21:04:07Z Shutdown: done```

**Comment by fanquake** [comment]
Thanks. As far as we are aware, this is a bug in macos Ventura, which has been confirmed as fixed in the next release (13.1).

Duplicate of #26455.

**Comment by asylejmani** [comment]
Thanks for the prompt response @fanquake.  Wasn't aware of that issue (I tried searching but I used F_FULLFSYNC 🤦)
